### PR TITLE
Return timestamped trajectory execution data

### PR DESCRIPTION
**Pull request by charleslin4**
- Return timestamped trajectories and force-torque readings from `ExecuteServoTrajectory`
- Rewrite `DensoController` to use `std::array` instead of `std::vector`
  - Reduces heap memory allocations during servo trajectory execution loop
  - Overall, reduces unnecessary heap usage

### CODE REVIEW

File: medra_pybind/src/DensoController.cpp [hunk 1]
```diff
@@ -564,6 +568,9 @@ namespace denso_controller
             vntPose = VNTFromRadVector(joint_position);
             hr = write_driver.SlvMove(&vntPose, &vntReturn);
 
+            auto now = std::chrono::system_clock::now();
+            joint_positions.push_back({now, joint_position});
+
             if (FAILED(hr))
```
Comment (hashb) [inline medra_pybind/src/DensoController.cpp:574]: we can end up losing some data if we throw an exception. we should try to capture as much data as possible particularly during failure

Comment (hashb) [inline medra_pybind/src/DensoController.cpp:574]: Also I think we are not closing the force sensing thread on failures, potential leak

Comment (charleslin4) [reply medra_pybind/src/DensoController.cpp]: No longer raising exceptions

File: medra_pybind/src/DensoController.cpp [hunk 2]
```diff
@@ -541,15 +546,17 @@ namespace denso_controller
...
+                    TimestampedForceSequence()
                 };
         }
 
         // Execute the trajectory
-        // BCAP_VARIANT vntPose, vntReturn;
         CommandServoJointResult result;
         bool force_limit_exceeded = false;
         bool trajectory_stopped_early = false;
+        TimestampedTrajectory joint_positions;
```
Comment (hashb) [inline medra_pybind/src/DensoController.cpp:559]: as discussed: reserve size here and dont use push back

**Review by hashb — APPROVED** [review]
lgtm

**Review by hashb — APPROVED** [review]
lgtm

**Pull request merged**